### PR TITLE
optimize zookeper calc + enable winpanda for dynamic discovery 

### PR DESCRIPTION
**Pull request by sergiimatusEPAM**
## High-level description

1)Winpanda handles the case when discovery_type is static. Once such set to "master_http_loadbalancer"(aka dynamic), setup command fails. The goal is to set such for dynamic case.
2) Zookeeper calc happens in Winpanda, therefore the PR to move the calculation to calc.py
3) cleanup of none-used calculations with master_list in Winpanda. + Test fixes from @mediapills 

## Corresponding DC/OS tickets (required)

  - [D2IQ-66830](https://jira.d2iq.com/browse/D2IQ-66830) set master for dynamic discovery_type
  - [D2IQ-66243](https://jira.d2iq.com/browse/D2IQ-66243) move mesos zookeeper calc into calc.py

## Related tickets (optional)

  - [D2IQ-ID](https://jira.mesosphere.com/browse/D2IQ-<number>) JIRA title / short description.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

File: gen/__init__.py [hunk 1]
```diff
@@ -492,6 +492,7 @@ def get_dcosconfig_source_target_and_templates(
         'package_ids',
         'template_filenames',
         'enable_windows_agents',
+        'zk_address',  # TODO needs to be relocated in config yaml file as a placeholder
```
Comment (sergiimatusEPAM) [inline gen/__init__.py:495]: Hi @jongiddy 
Need your advice on dilemma disccovered. In order to populate new variable `zk_address`, it needs to be added either in 
1) here at get_dcosconfig_source_target_and_templates() directly 
or 
2) dcos-config.yaml.
I and @mediapills believe that 2nd would be more logical, however, where is to allocate {{ zk_address }} in dcos-config.yaml exactly, as the parameter will be NOT used by any of the configs, in-fact. It requires for mesos-agent.exe startup command ONLY which is picked from nssm config, and has no relation with dcos-config.yaml
See example, please: 
```
mesos-agent.exe --master=zk://{{ zk_address }}/mesos

#static
mesos-agent.exe --master=zk://172.16.0.1:2181,172.16.0.2:2181,172.16.0.3:2181/mesos
#dynamic
mesos-agent.exe --master=zk://int-sm-ee-d2iq-66812-1-7d0e3291b8692e6e.elb.us-east-1.amazonaws.com:2181/mesos
```

**Comment by sergiimatusEPAM** [comment]
@mesosphere-mergebot request-review @mediapills

File: gen/__init__.py (hunk 1 above)
Comment (jongiddy) [reply gen/__init__.py:495]: Following our discussion, preferably, the generation of the variable is in `calc.py` and, even if the variable is not used apart from in the Mesos package, we can force the variable to be calculated by creating a dummy file in `dcos-config-windows.yaml`, as shown in my previous attempt at this problem: https://github.com/dcos/dcos/pull/7101/files#diff-3a176d7226c4a814bad89c707c845253

Comment (sergiimatusEPAM) [reply gen/__init__.py]: @jongiddy recommended to workaround the need by using dummy.txt at Windows template.

**Comment by sergiimatusEPAM** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by sergiimatusEPAM** [comment]
@mesosphere-mergebot request-review @jongiddy

**Comment by sergiimatusEPAM** [comment]
@mesosphere-mergebot changelog-not-required Windows GA re-factoring

**Pull request merged**